src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -271,6 +270,33 @@ public ValidatedAddress ValidateAddress(string address)
             return res;
         }
 
+        /// <summary>
+        /// RPC method for returning a block.
+        /// Currently only supports raw (hex) format, Json format is not yet supported.
```

Comment:
I think that, where possible, the RPC call could just invoke the underlying API methods to avoid the possibility of code duplication. That would, probably, solve the JSON issue, as the `BlockStore` controller appears to already implement this logic with JSON support in the `block` endpoint.

Reply:
Do we already do that? (call the api endpoints from an RPC endpoint)

Reply:
No I don't think that is the current design. It is a good point of discussion though whether we should be.  Should RPC sit on top of API? I'm a bit concerned about all the coupling between RPC, API, and Features.

Reply:
I think my main problem with that is that we then need two sets of tests for everything. But the coupling is problematic, yes.

I suppose it is an unnecessary concern if the API methods themselves diverge significantly in functionality from the RPC versions in future.

Reply:
If we don't already do that then leave this as is, we will indeed have too many shared code between the two, if we can though we use a Service that both API and RPC can call for the lower level stuff (when too much logic is inside the controller)